src/pass/autodiff.h

```diff
@@ -0,0 +1,146 @@
+/*!
```

Comment:
The current location of `autodiff.h` prevents it from being used by standalone C++ applications since they don't have an access to src folder. Could we move this file to `tvm/include` ?